go.mod

```diff
@@ -20,9 +19,12 @@ require (
 	k8s.io/api v0.31.0
 	k8s.io/apimachinery v0.31.0
 	k8s.io/client-go v0.31.0
+	k8s.io/api v0.31.0
+	k8s.io/apimachinery v0.31.0
+	k8s.io/client-go v0.31.0
 	k8s.io/klog/v2 v2.130.1
 	k8s.io/utils v0.0.0-20240711033017-18e509b52bc8
-	sigs.k8s.io/controller-runtime v0.19.0
+	sigs.k8s.io/controller-runtime v0.18.4
```

Comment:
this commit contains multiple downgrades, is that intentional?

Reply:
am not sure if I did that probably dependabot did it? or while rebasing it got missed?
earlier I saw CI failures for incompatibility with `controller-runtime v0.18.4`, and I updated it to `v0.19.0` in my last commit.

let me squash the last commit